File: 2262-solving-questions-with-brainpower/2262-solving-questions-with-brainpower.cpp
class Solution {
public:
    long long mostPoints(vector<vector<int>>& questions) {
        int n=questions.size();
        vector<long long>dp(n+5,0);
        for(int i=n-1;i>=0;i--){
            dp[i]=max(dp[i+1],questions[i][0]+dp[min(n+1,i+questions[i][1]+1)]);
        }
        return *max_element(dp.begin(),dp.end());
    }
};

File: 2262-solving-questions-with-brainpower/2262-solving-questions-with-brainpower.js
var mostPoints = function(nums) {
    let n=nums.length;
    let dp=new Array(n+1);
    dp.fill(0);
    for(let i=n-1;i>=0;i--){
        dp[i]=Math.max(dp[i+1],nums[i][0]+dp[Math.min(i+nums[i][1]+1,n)]);
    }
    return dp[0];
};